plugins/modules/object_keys.py

```diff
@@ -128,32 +144,138 @@ def _get_key_by_label(self, label: str) -> Optional[ObjectStorageKeys]:
             )
 
     def _create_key(
-        self, label: str, bucket_access: Union[dict, List[dict]]
+        self,
+        label: str,
+        bucket_access: Optional[List[Dict[str, Any]]],
+        regions: Optional[List[str]],
     ) -> Optional[ObjectStorageKeys]:
         """Creates an Object Storage key with the given label and access"""
 
+        # The API will reject explicit null values for `bucket_access.region`
+        if bucket_access is not None:
+            bucket_access = [
+                filter_null_values(grant) for grant in bucket_access
+            ]
+
         try:
             return self.client.object_storage.keys_create(
-                label, bucket_access=bucket_access
+                label, bucket_access=bucket_access, regions=regions
             )
         except Exception as exception:
             return self.fail(
                 msg="failed to create object storage key: {0}".format(exception)
             )
 
+    @staticmethod
+    def _access_changed(key: ObjectStorageKeys, params: Dict[str, Any]) -> bool:
```

Comment:
This method contains a lot of logic just to raise an error, but I really wanted to make sure users would be informed if they attempted to update the bucket access configuration of a key.